File: edu-client/lib/features/approval/models/approval_request.dart
class ApprovalRequest {
  final int id;
  final String title;
  final String lawReference;
  final String aiGeneratedMarkdown;
  final String validationDetails;
  final double hallucinationScore;
  final String status; // PENDING, APPROVED, REJECTED
  final DateTime createdAt;

  const ApprovalRequest({
    required this.id,
    required this.title,
    required this.lawReference,
    required this.aiGeneratedMarkdown,
    required this.validationDetails,
    required this.hallucinationScore,
    required this.status,
    required this.createdAt,
  });

  factory ApprovalRequest.fromJson(Map<String, dynamic> json) {
    return ApprovalRequest(
      id: json['id'] as int,
      title: json['title'] as String? ?? 'N/A',
      lawReference: json['lawReference'] as String? ?? 'N/A',
      aiGeneratedMarkdown: json['aiGeneratedMarkdown'] as String? ?? '',
      validationDetails: json['validationDetails'] as String? ?? '',
      hallucinationScore: (json['hallucinationScore'] as num? ?? 0.0).toDouble(),
      status: json['status'] as String? ?? 'PENDING',
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt'] as String) 
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'lawReference': lawReference,
      'aiGeneratedMarkdown': aiGeneratedMarkdown,
      'validationDetails': validationDetails,
      'hallucinationScore': hallucinationScore,
      'status': status,
      'createdAt': createdAt.toIso8601String(),
    };
  }

  ApprovalRequest copyWith({
    int? id,
    String? title,
    String? lawReference,
    String? aiGeneratedMarkdown,
    String? validationDetails,
    double? hallucinationScore,
    String? status,
    DateTime? createdAt,
  }) {
    return ApprovalRequest(
      id: id ?? this.id,
      title: title ?? this.title,
      lawReference: lawReference ?? this.lawReference,
      aiGeneratedMarkdown: aiGeneratedMarkdown ?? this.aiGeneratedMarkdown,
      validationDetails: validationDetails ?? this.validationDetails,
      hallucinationScore: hallucinationScore ?? this.hallucinationScore,
      status: status ?? this.status,
      createdAt: createdAt ?? this.createdAt,
    );
  }
}

File: edu-server/src/main/java/com/everlaw/edu/domain/approval/ApprovalRequest.java
package com.everlaw.edu.domain.approval;

import com.everlaw.edu.domain.curriculum.Curriculum;
import com.everlaw.edu.domain.lesson.Lesson;
import com.everlaw.edu.global.domain.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "approval_requests", indexes = {
        @Index(name = "idx_approval_requests_status", columnList = "status"),
        @Index(name = "idx_approval_requests_lesson", columnList = "lesson_id")
})
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ApprovalRequest extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // 신규 교안 생성의 경우 최종 승인 전까지 Lesson이 없으므로 Nullable 허용
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lesson_id", nullable = true)
    @com.fasterxml.jackson.annotation.JsonIgnore
    private Lesson lesson;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "curriculum_id", nullable = false)
    @com.fasterxml.jackson.annotation.JsonIgnore
    private Curriculum curriculum;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(name = "law_reference", nullable = false, length = 150)
    private String lawReference;

    @Column(name = "law_reference_body", columnDefinition = "TEXT")
    private String lawReferenceBody;

    @Column(name = "ai_generated_markdown", nullable = false, columnDefinition = "TEXT")
    private String aiGeneratedMarkdown;

    @Column(name = "quiz_payload", columnDefinition = "TEXT")
    private String quizPayload;

    @Column(name = "validation_details", columnDefinition = "TEXT")
    private String validationDetails;

    @Column(name = "hallucination_score", nullable = false)
    private Double hallucinationScore;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ApprovalStatus status;

    // 동시성 제어 및 중복 승인 레이스 컨디션 방지를 위한 낙관적 락 버전 필드
    @Version
    private Long version;

    @Builder
    public ApprovalRequest(Lesson lesson, Curriculum curriculum, String title, String lawReference, String lawReferenceBody,
                           String aiGeneratedMarkdown, String quizPayload, String validationDetails, Double hallucinationScore,
                           ApprovalStatus status) {
        this.lesson = lesson;
        this.curriculum = curriculum;
        this.title = title;
        this.lawReference = lawReference;
        this.lawReferenceBody = lawReferenceBody;
        this.aiGeneratedMarkdown = aiGeneratedMarkdown;
        this.quizPayload = quizPayload;
        this.validationDetails = validationDetails;
        this.hallucinationScore = hallucinationScore;
        this.status = status != null ? status : ApprovalStatus.PENDING;
    }

    public void approve(Lesson approvedLesson) {
        if (this.status != ApprovalStatus.PENDING) {
            throw new IllegalStateException("Only PENDING requests can be approved.");
        }
        this.status = ApprovalStatus.APPROVED;
        this.lesson = approvedLesson;
    }

    public void reject() {
        if (this.status != ApprovalStatus.PENDING) {
            throw new IllegalStateException("Only PENDING requests can be rejected.");
        }
        this.status = ApprovalStatus.REJECTED;
    }

    public void updateRequest(String title, String lawReferenceBody, String aiGeneratedMarkdown, 
                              String quizPayload, String validationDetails, Double hallucinationScore) {
        this.title = title;
        this.lawReferenceBody = lawReferenceBody;
        this.aiGeneratedMarkdown = aiGeneratedMarkdown;
        this.quizPayload = quizPayload;
        this.validationDetails = validationDetails;
        this.hallucinationScore = hallucinationScore;
        this.status = ApprovalStatus.PENDING;
    }
}
